vhdbuilder/packer/buildperformance/pkg/service/kusto_functions.go

```diff
@@ -0,0 +1,63 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-kusto-go/kusto"
+	kustoErrors "github.com/Azure/azure-kusto-go/kusto/data/errors"
+	"github.com/Azure/azure-kusto-go/kusto/data/table"
+	"github.com/Azure/azure-kusto-go/kusto/ingest"
+	"github.com/Azure/azure-kusto-go/kusto/kql"
+)
+
+func CreateKustoClient(kustoEndpoint string, kustoClientId string) (*kusto.Client, error) {
+	kcsb := kusto.NewConnectionStringBuilder(kustoEndpoint).WithUserManagedIdentity(kustoClientId)
+	client, err := kusto.New(kcsb)
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+func IngestData(client *kusto.Client, ctx context.Context, kustoDatabase string, kustoTable string, buildPerformanceDataFile string, kustoIngestionMap string) error {
```

Comment:
nit: we usually have `ctx context.Context` as the first positional argument for funcs that take in a context

Reply:
Fixed.